bellows/uart.py

```diff
@@ -105,13 +107,16 @@ def rst_frame_received(self, data):
 
     def rstack_frame_received(self, data):
         """Reset acknowledgement frame receive handler"""
-        LOGGER.debug("RSTACK frame: %s", binascii.hexlify(data))
         self._send_seq = 0
         self._rec_seq = 0
-        if self._reset_future is None:
-            LOGGER.warn("Reset future is None")
-            return
-        self._reset_future.set_result(True)
+        code = t.NcpResetCode(data[2])
```

Comment:
Hrm. I've just looked at the docs again, and I see that it says "> 0x80: EM3XX-specific error reset code". I wonder if there's a way that we can make this code safe for other reset codes, rather than throwing an exception.

I haven't thought about this too hard, but I think

```python3
try:
    code = t.NcpResetCode(data[2])
except:
    code = t.NcpResetCode.RESET_UNKNOWN_REASON
```

What do you think?